sphinx/directives/code.py

```diff
@@ -332,6 +309,31 @@ def run(self):
                     res.append(line)
             lines = res
 
+        linespec = self.options.get('lines')
+        if linespec:
+            try:
+                linelist = parselinenos(linespec, len(lines))
+            except ValueError as err:
+                return [document.reporter.warning(str(err), line=self.lineno)]
+
+            if 'lineno-match' in self.options:
+                # make sure the line list is not "disjoint".
+                previous = linelist[0]
+                for line_number in linelist[1:]:
+                    if line_number == previous + 1:
+                        previous = line_number
+                        continue
+                    return [document.reporter.warning(
+                        'Cannot use "lineno-match" with a disjoint set of '
+                        '"lines"', line=self.lineno)]
+                linenostart += linelist[0]
+            # just ignore non-existing lines
+            lines = [lines[i] for i in linelist if i < len(lines)]
+            if not lines:
+                return [document.reporter.warning(
+                    'Line spec %r: no lines pulled from include file %r' %
+                    (linespec, filename), line=self.lineno)]
+
```

Comment:
in case both lineno-start and lineno-match are in use, the earlier code did not handle all branches the same, now lineno-match makes sure always linenostart gets default value 1